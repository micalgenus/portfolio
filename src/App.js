import React, { Component } from 'react';
import Layout from '@/Containers/Layout';

import Router from '@/Router';

class App extends Component {
  render() {
    return <Layout {...this.props} routers={Router} />;
  }
}

export default App;
