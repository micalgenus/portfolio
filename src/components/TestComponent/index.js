import React from 'react';
import PropTypes from 'prop-types';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

export default class TestComponent {
  constructor(props) {
    this.props = {
      name: props.name,
      component: props.namcomponente,
      snapshot: props.snapshot === true ? true : false,
      args: props.args || {},
    };
  }

  run = callback => {
    const component = this.props.component;
    const args = this.props.args;

    describe(this.props.name, () => {
      if (this.props.snapshot === true) {
        it('snapshot test', () => {
          const wrapper = shallow(<component {...args} />);
          expect(wrapper).toMatchSnapshot();
        });
      }

      if (typeof callback === 'function') callback.bind(this)();
    });
  };
}

TestComponent.defaultProps = { snapshot: false, args: {} };

TestComponent.propTypes = {
  name: PropTypes.string,
  component: PropTypes.object,
  snapshot: PropTypes.bool,
  args: PropTypes.object,
};

const snapshotTest = (name, component, argv) => {
  if (!name || !component) return null;
  const snapshotTestComponent = new TestComponent({ name: name, component: component, snapshot: true, args: argv });
  snapshotTestComponent.run(null);
};

export { snapshotTest };
