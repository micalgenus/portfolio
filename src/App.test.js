import React from 'react';

import Stores from '@/Stores';

import { snapshotTest } from '@/TestComponent';

import App from './App';

snapshotTest({
  name: '/App',
  component: (
    <Stores>
      <App />
    </Stores>
  ),
});
