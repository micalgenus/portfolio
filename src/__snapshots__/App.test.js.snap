// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/App snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default>
    <App />
  </_default>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <App />,
      "page": PageStore {
        "currentPath": "/",
        "loadingPages": Array [],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "PageStore@2.keys",
            "observers": Set {},
          },
          "name": "PageStore@2",
          "target": [Circular],
          "values": Map {
            "currentPath" => "/",
            "loadingPages" => Array [],
          },
        },
      },
      "scroll": ScrollStore {
        "height": 0,
        "movedScrollY": 0,
        "scrollY": 0,
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "ScrollStore@3.keys",
            "observers": Set {},
          },
          "name": "ScrollStore@3",
          "target": [Circular],
          "values": Map {
            "scrollY" => 0,
            "movedScrollY" => 0,
            "height" => 0,
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <App />,
        "page": PageStore {
          "currentPath": "/",
          "loadingPages": Array [],
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "PageStore@2.keys",
              "observers": Set {},
            },
            "name": "PageStore@2",
            "target": [Circular],
            "values": Map {
              "currentPath" => "/",
              "loadingPages" => Array [],
            },
          },
        },
        "scroll": ScrollStore {
          "height": 0,
          "movedScrollY": 0,
          "scrollY": 0,
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "ScrollStore@3.keys",
              "observers": Set {},
            },
            "name": "ScrollStore@3",
            "target": [Circular],
            "values": Map {
              "scrollY" => 0,
              "movedScrollY" => 0,
              "height" => 0,
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
